### PR TITLE
Support transaction search by Subscription number

### LINKED ISSUE #867: Make Transaction List View searchable by the subscription ID (2/3)
paJDYF-O8-p2

Related: #866 #868

### PR DESCRIPTION
**Pull request by dechov**
Fixes #867

#### Changes proposed in this Pull Request

Suggests and supports "Subscription #xxxx" terms in the Transaction search box, filtering the view down to transactions associated with orders that are related to the given subscription.

subscription-search-1

subscription-search-2

subscription-search-3

(Note: the visual glitch in the latter screenshot appears for me in other WooCommerce Admin search fields, and I intend to propose a fix upstream for that and a couple of other minor things.)

https://github.com/Automattic/woocommerce-payments/commit/ce0c4fefac12a17a8427301c836d750a5ed479ef fixes the order term prefix association for the translated case.

#### Testing instructions

- With Subscriptions having been purchased via WCPay…
- Type a Subscription number into the search field and select the corresponding term suggestion
  - Alternatively, type "Subscription #xxxx" as a free text term
- Observe that only those transactions having an association with the queried subscription now appear

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payments-utils.php [hunk 1]
```diff
@@ -149,51 +149,62 @@ public static function mark_payment_expired( $order ) {
...
-			return '';
-		}
-
-		$order = wc_get_order( $order_id );
 
-		if ( ! $order ) {
-			return '';
+		$subscription_term = __( 'Subscription #', 'woocommerce-payments' );
+		if ( function_exists( 'wcs_get_subscription' ) && substr( $term, 0, strlen( $subscription_term ) ) === $subscription_term ) {
+			$term_parts      = explode( '#', $term, 2 );
```
Comment (dechov) [inline includes/class-wc-payments-utils.php:172]: This block is adapted from the "Order #" term handling above.

**Comment by dechov** [comment]
Still needs changes to automated tests.

File: includes/class-wc-payments-utils.php [hunk 2]
```diff
@@ -149,51 +149,62 @@ public static function mark_payment_expired( $order ) {
...
+					function ( $order ) {
+						return $order->get_meta( '_charge_id' );
+					},
+					$subscription->get_related_orders( 'all' )
+				);
+			}
 		}
 
-		return $order->get_meta( '_charge_id' );
+		return null;
```
Comment (dwainm) [inline includes/class-wc-payments-utils.php:185]: Do you think we could opt for an empty array instead? The simplifies the function to one type.

Comment (dechov) [reply includes/class-wc-payments-utils.php]: 👍 Done in https://github.com/Automattic/woocommerce-payments/pull/936/commits/eb3823d630870d2a55c6c51683ffd77cbfc50cd4

**Review by DanReyLop — APPROVED** [review]
Code looks good and works as advertised.

:shipit:

**Comment by dechov** [comment]
Thanks for the review @DanReyLop!

> propose a fix upstream for that and a couple of other minor things

Opened https://github.com/woocommerce/woocommerce-admin/pull/5199.

**Pull request merged**